TeamCode/src/main/java/org/firstinspires/ftc/teamcode/teleop/TeleOp.java

```diff
@@ -0,0 +1,238 @@
+    double leftFPower;
+    double rightFPower;
+    double leftBPower;
+    double rightBPower;
+    double drive;
+    double turn;
+    double strafe;
+    double intakePower = 0.5;
+    private final double armStartPos = 0.0;
+    private final double armTargetPos = 0.0;
+    private final double armTolerance = 2.5;
+    private final double viperSlideStartPos = 0.0;
+    private final double viperSlideTargetPos = 0.0;
+    private final double jointStartPos = 0.0;
+    private final double jointTargetPos = 0.0;
+
+    @Override
+    public void init() {
+        telemetry.addData("Status", "Initializing");
+
+        // Do all init stuff.
+        robot = new Hardware(hardwareMap, telemetry);
+
+        // Init Servos.
+        robot.introduce(new HardwareElement<>(Servo.class, hardwareMap, "clawServo"));
+        robot.introduce(new HardwareElement<>(CRServo.class, hardwareMap, "jointServo"));
+
+        // Init CR Servos.
+        robot.introduce(new HardwareElement<>(CRServo.class, hardwareMap, "geekoWheelCRServo"));
+        robot.introduce(new HardwareElement<>(CRServo.class, hardwareMap, "tubeCRServo"));
+
+        // Init DcMotors.
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "leftFront", "setDirection:FORWARD"));
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "leftRear", "setDirection:FORWARD"));
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "rightFront", "setDirection:FORWARD"));
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "rightBack", "setDirection:FORWARD"));
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "armMotor", "setMode(DcMotor.RunMode.RUN_USING_ENCODER)"));
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "leftViperSlideMotor", "setMode(DcMotor.RunMode.RUN_USING_ENCODER)"));
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "rightViperSlideMotor", "setMode(DcMotor.RunMode.RUN_USING_ENCODER)"));
+
+        telemetry.addData("Status", "Initialized");
+    }
+
+    @Override
+    public void loop() {
+
+        // Values for drive.
+        drive = gamepad1.left_stick_y * 0.8;
+        turn = -gamepad1.left_stick_x * 0.6;
+        strafe = gamepad1.right_stick_x * 0.8;
+
+        // Set power to values calculated above.
+        robot.<DcMotor>get("leftFront").setPower(leftFPower);
+        robot.<DcMotor>get("leftRear").setPower(leftBPower);
+        robot.<DcMotor>get("rightFront").setPower(rightFPower);
+        robot.<DcMotor>get("rightRear").setPower(rightBPower);
+
+        telemetry.addData("leftFPower: ", leftFPower);
+        telemetry.addData("leftBPower: ", leftBPower);
+        telemetry.addData("rightFPower: ", rightFPower);
+        telemetry.addData("rightBPower: ", rightBPower);
+
+        // Calling scoring() via left bumper and resetting via left trigger
+        if (gamepad1.left_bumper) {
```

Comment:
might be smart to add some sort of safety net that makes it so resetPos() won't run at the same time as scoring.